src/archemist/core/optimisation/recipe_generator.py

```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+import logging
+import re
+from typing import Dict
+
+class RecipeGenerator:
+    def __init__(self, template_path, recipe_path) -> None:
+        self._recipe_dict  = YamlHandler.load_recipe_file(template_path)
+        self._recipe_dir = recipe_path
+        self._placeholder_keys_all = {}
+        self._placeholder_keys_any = {}
+        self._find_placeholders(self._recipe_dict)
+
+    # Methods
+    def generate_recipe(self, optimised_parameters_data):
+        optimised_parameters_dict = optimised_parameters_data.to_dict()
+        if self._placeholder_keys_any:
+            self._recipe_dict = self._update_recipe(self._recipe_dict,'any', optimised_parameters_dict)
```

Comment:
you don't want to overwrite the self._recipe_dict as it contains your recipe template. You should just create a new dict and return it out of this method.